libfwupdplugin/fu-device.c

```diff
@@ -1468,14 +1472,13 @@ fu_device_set_version_lowest (FuDevice *self, const gchar *version, FwupdVersion
  * fu_device_set_version_bootloader:
  * @self: A #FuDevice
  * @version: (allow-none): a string, e.g. `1.2.3`
- * @fmt: a #FwupdVersionFormat, e.g. %FWUPD_VERSION_FORMAT_TRIPLET
  *
  * Sets the device bootloader version, sanitizing the string if required.
  *
  * Since: 1.3.9
  **/
 void
-fu_device_set_version_bootloader (FuDevice *self, const gchar *version, FwupdVersionFormat fmt)
+fu_device_set_version_bootloader (FuDevice *self, const gchar *version)
```

Comment:
I'm almost wondering if it's worth having a static function with an enum or something for the type of version string being passed so this isn't duplicated 90% 3 times.

Reply:
I think if it gets any more complicated I agree, but at the moment it's pretty self contained.